SwiftGenerics.playground/Contents.swift

```diff
@@ -0,0 +1,99 @@
+import UIKit
+
+struct CountedSet<Element: Hashable>: ExpressibleByArrayLiteral {
+    
+    private(set) var elements = [Element: Int]()
+    
+    init(arrayLiteral elements: Element...) {
+        for element in elements {
+            insert(element: element)
+        }
+    }
+    
+    mutating func insert(element: Element) {
+        if let _ = elements[element] {
+            elements[element]! += 1
+        } else {
+            elements[element] = 1
+        }
+    }
+    
+    mutating func remove(element: Element) {
+        guard let _ = elements[element] else { return }
+        elements[element]! -= 1
+        
+        if elements[element] == 0 {
+            elements.removeValue(forKey: element)
+        }
```

Comment:
This is another opportunity to keep the value around so you don't need to fetch it more than once